ng-dev/release/publish/test/release-notes/generation.spec.ts

```diff
@@ -0,0 +1,145 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import {installSandboxGitClient, SandboxGitClient} from '../test-utils/sandbox-git-client';
+import {mkdirSync, rmdirSync} from 'fs';
+import {testTmpDir} from '../test-utils/action-mocks';
+import {getMockGitClient} from '../test-utils/git-client-mock';
+import {GithubConfig} from '../../../../utils/config';
+import {SandboxGitRepo} from '../test-utils/sandbox-testing';
+import {ReleaseNotes} from '../../../notes/release-notes';
+import {ReleaseConfig} from '../../../config';
+import {changelogPattern, parse} from '../test-utils/test-utils';
+
+describe('release notes generation', () => {
+  let releaseConfig: ReleaseConfig;
+  let githubConfig: GithubConfig;
+  let client: SandboxGitClient;
+
+  beforeEach(() => {
+    // Clear the temporary directory used by the sandbox git client. We do not want
+    // the repo state to persist between tests.
+    rmdirSync(testTmpDir, {recursive: true});
```

Comment:
Should we perform this clean of the temp directory as part of the `getMockGitClient` function, if I remember correctly this is duplicated in another place where `getMockClient` is used.

Reply:
I have looked into this as well; but couldn't come up with a good way to de-duplicate this without "hiding stuff". Doing it inside `getMockGitClient` seems unexpected because I could totally call this method twice but then I'd not expect it to clear the whole temporary directory. Also it's not specified that `getMockGitClient` necessarily runs at the beginning of a test right? That's why it currently exists here in `beforeEach` and for the release action tests within the `setup<..>Action` test function.

Reply:
Hmm okay, lets leave it as is then and we can keep it in mind for if we find a way in the future.

Reply:
yeah, happy to change if we can come up with something better. I think at this point it's duplicated, but more clear than doing the cleanup somewhere inside. (note I have also considered doing it inside `SandboxGitClient.createInstance`)